vyked/registry.py

```diff
@@ -216,7 +216,17 @@ def receive(self, packet: dict, protocol, transport):
         elif request_type == 'pong':
             self._ping(packet)
         elif request_type == 'ping':
-            self._pong(packet, protocol)
+            if 'payload' in packet:
```

Comment:
def _handle_ping(self, packet, protocol):
        """ Responds to pings from registry_client only if the node_ids present in the ping payload are registered

        :param packet: The 'ping' packet received
        :param protocol: The protocol on which the pong should be sent
        """
        if 'payload' in packet:
            to_send=True
            node_ids = list(packet['payload'].values())
            for node_id in node_ids:
                if self._repository.get_node(node_id) is None:
                    to_send = False
                    break
            if to_send:
                self._pong(packet, protocol)
        else:
            self._pong(packet, protocol)